akka-http2-support/src/test/scala/akka/http/impl/engine/http2/FramesInFramesOutSpec.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.impl.engine.http2
+
+import akka.NotUsed
+import akka.http.impl.engine.http2.Http2Protocol.ErrorCode
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.http2.Http2StreamIdHeader
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{ BidiFlow, Flow, Keep }
+import akka.stream.testkit.scaladsl.{ TestSink, TestSource }
+import akka.stream.testkit.{ TestPublisher, TestSubscriber }
+import akka.testkit.AkkaSpec
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.duration._
+
+class FramesInFramesOutSpec extends AkkaSpec with Eventually {
```

Comment:
Would it make sense to convert Http2ServerSpec, instead, to not rely on network bytes but parsed frames instead? Otherwise, how do we determine which tests will live where? The good thing about Http2ServerSpec is that it really checks the whole stack (without TLS) and already has most of the utilities to render and parse frames to ByteStrings.